src/compas_viewer/scene/graphobject.py

```diff
@@ -26,31 +30,17 @@ def __init__(self, graph: Graph, **kwargs):
         super().__init__(graph=graph, **kwargs)
         self.graph: Graph
 
-    def _read_points_data(self) -> DataType:
-        positions = []
-        colors = []
-        elements = []
-        i = 0
-
-        for node in self.graph.nodes():
-            positions.append(self.graph.node_coordinates(node))
-            colors.append(self.pointscolor.get(node, self.pointscolor["_default"]))  # type: ignore
-            elements.append([i])
-            i += 1
-        return positions, colors, elements
-
-    def _read_lines_data(self) -> DataType:
-        positions = []
-        colors = []
-        elements = []
-        i = 0
-
-        for u, v in self.graph.edges():
-            color = self.linescolor.get((u, v), self.linescolor["_default"])  # type: ignore
-            positions.append(self.graph.node_coordinates(u))
-            positions.append(self.graph.node_coordinates(v))
-            colors.append(color)
-            colors.append(color)
-            elements.append([i + 0, i + 1])
-            i += 2
-        return positions, colors, elements
+    @property
+    def points(self) -> Optional[list[Point]]:
+        """The points to be shown in the viewer."""
+        return [Point(*self.graph.node_coordinates(node)) for node in self.graph.nodes()]
+
+    @property
+    def lines(self) -> Optional[list[Line]]:
+        """The lines to be shown in the viewer."""
+        return [Line(self.graph.node_coordinates(u), self.graph.node_coordinates(v)) for u, v in self.graph.edges()]
+
+    @property
+    def surfaces(self) -> Optional[list[Tuple[Point, Point, Point]]]:
+        """The surface to be shown in the viewer. Currently only triangles are supported."""
+        pass
```

Comment:
if the behaviour is based on an implicit None return, i would explicitly return None, instead of passing...